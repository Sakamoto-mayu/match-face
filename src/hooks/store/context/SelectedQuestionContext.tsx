"use client";
import {
  Dispatch,
  ReactNode,
  Reducer,
  createContext,
  useContext,
  useReducer,
} from "react";
import { selectedQuestionReducer } from "../reducer/selectedQuestionReducer";
import { SelectReducerAction } from "@/types/admin/tasks/register/types";

const SelectedQuestionContext = createContext<
  [string[], Dispatch<SelectReducerAction>] | undefined
>(undefined);

export const SelectedQuestionProvider = ({ children }: { children: ReactNode }) => {
  const [state, dispatch] = useReducer<Reducer<string[], SelectReducerAction>>(
    selectedQuestionReducer,
    []
  );
  return (
    <SelectedQuestionContext.Provider value={[state, dispatch]}>
      {children}
    </SelectedQuestionContext.Provider>
  );
};

export const useSelectQuestion = () => {
  const context = useContext(SelectedQuestionContext);
  if (context === undefined)
    throw new Error("Function 'useUserSelect' must be used in Provider.");
  return context;
};
