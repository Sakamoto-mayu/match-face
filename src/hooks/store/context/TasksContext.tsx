"use client";

import React from "react";

// 職種フィルター用context
const JobsFilterContext = React.createContext<
  [string[], React.Dispatch<React.SetStateAction<string[]>>] | undefined
>(undefined);

export const JobsFilterProvider = ({
  children,
}: {
  children: React.ReactNode;
}) => {
  const [filterlist, setFilterList] = React.useState<string[]>([]);
  return (
    <JobsFilterContext.Provider value={[filterlist, setFilterList]}>
      {children}
    </JobsFilterContext.Provider>
  );
};

export const useJobsFilter = () => {
  const context = React.useContext(JobsFilterContext);
  if (context === undefined) {
    throw new Error("useRefine must be used within a Provider");
  }
  return context;
};
// ここまで
