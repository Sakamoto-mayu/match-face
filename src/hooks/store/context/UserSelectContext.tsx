"use client";
import {
  Dispatch,
  ReactNode,
  Reducer,
  createContext,
  useContext,
  useReducer,
} from "react";
import { userSelectReducer } from "../reducer/userSelectReducer";
import { UserSelectReducerAction } from "@/types/admin/tasks/register/types";

const UserSelectContext = createContext<
  [string[], Dispatch<UserSelectReducerAction>] | undefined
>(undefined);

export const CounterProvider = ({ children }: { children: ReactNode }) => {
  const [state, dispatch] = useReducer<Reducer<string[], UserSelectReducerAction>>(
    userSelectReducer,
    []
  );
  return (
    <UserSelectContext.Provider value={[state, dispatch]}>
      {children}
    </UserSelectContext.Provider>
  );
};

export const useUserSelect = () => {
  const context = useContext(UserSelectContext);
  if (context === undefined)
    throw new Error("Function 'useCounter' must be used in Provider.");
  return context;
};
