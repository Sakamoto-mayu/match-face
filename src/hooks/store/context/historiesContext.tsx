'use client'
import { createContext, useContext } from "react";
import { Departments, Skills } from "@/types/admin/histories/admin_histories";
import { skills,departments } from "@/const/admin_histories";

export const SkillsContext = createContext<Skills|undefined>(undefined)
export const DepartmentsContext = createContext<Departments|undefined>(undefined)

// constからのimport(valueの値)をfetchで取得したデータに変える
// const skills = fetch('')
// const departments = fetch('')

export function HistoriesProvider({children}:{children:React.ReactElement}) {
    return (
        <SkillsContext.Provider value = {skills}>
        <DepartmentsContext.Provider value = {departments}>
            {children}
        </DepartmentsContext.Provider>
        </SkillsContext.Provider>
    )
}
export const useSkills = () => {
    const context = useContext(SkillsContext)
    if(context === undefined){
        throw new Error('skillsが見つかりません')
    }
    return context
}
export const useDepartments = () => {
    const context = useContext(DepartmentsContext)
    if(context === undefined){
        throw new Error('departmentsが見つかりません')
    }
    return context
}
