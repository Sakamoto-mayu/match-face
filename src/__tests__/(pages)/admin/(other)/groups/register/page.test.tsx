import {render,screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import GroupsRegisterPage from '@/app/(pages)/admin/(other)/groups/register/page'
import React from 'react'
import "@testing-library/jest-dom"
import { UserSelectProvider } from '@/hooks/store/context/UserSelectContext'

// useContextをモーダルで使用しているため、Providerを追加しています

describe('管理者/グループ設定画面のテスト',()=>{
    const user = userEvent.setup()
    beforeEach(()=>{
        render(
            <UserSelectProvider>
                <GroupsRegisterPage/>
            </UserSelectProvider>
        )
    })
    describe('スナップショットテスト',()=>{
        it('レンダリング時',async()=>{
            const view = render(
                <UserSelectProvider>
                    <GroupsRegisterPage/>
                </UserSelectProvider>
            )
            expect(view.container).toMatchSnapshot()
        })
    })
    describe('モーダル表示テスト',()=>{
        it('グループを設定するボタン押下で確認表示',async()=>{
            const registerButton = screen.getByTestId('registerConfirm')
            await user.click(registerButton)
            const modalConfirmation = screen.getByText('グループを設定してよろしいですか?')
            expect(modalConfirmation).toBeInTheDocument
        })
    })
    
    // 実装を進めたら続きを書きます
    // describe('グループ設定',()=>{
    //     it('設定するボタン押下で新規設定',async()=>{
    //         const registerButtonTrue =screen.getByTestId('registerTrue')
    //         await user.click(registerButtonTrue)
    //     })
    // })
})
