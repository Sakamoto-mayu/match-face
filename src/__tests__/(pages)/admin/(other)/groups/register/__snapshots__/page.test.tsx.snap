// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`管理者/グループ設定画面のテスト スナップショットテスト レンダリング時 1`] = `
<div>
  <div
    class="flex flex-col items-center justify-center h-screen"
  >
    <div
      class="border-2  py-12 px-12 "
    >
      <form>
        <label
          for="group_name"
        >
          ▶グループ名を設定する
        </label>
        <div>
          <input
            class="my-3 px-2 py-1 border-2 border-gray-300"
            id="group_name"
            style="width: 600px;"
          />
        </div>
        <br />
        <label
          for="user"
        >
          ▶ユーザーを選択する
        </label>
        <span>
           
        </span>
        <button
          class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm py-2 px-4 rounded-full border shadow-md text-xs"
        >
          追加
        </button>
        <div>
          <input
            class="my-3 px-2 py-1 border-2 border-gray-300"
            id="user"
            style="width: 600px;"
          />
        </div>
        <br />
        <label
          for="group_description"
        >
          ▶備考
        </label>
        <div>
          <input
            class="my-3 px-2 py-1 border-2 border-gray-300"
            id="group_description"
            style="width: 600px;"
          />
        </div>
      </form>
      <br />
      <div
        class="flex flex-col items-center justify-center"
      >
        <div>
          <button
            class="bg-orange drop-shadow-md hover:saturate-150 active:drop-shadow-none active:shadow-inner hover:bg-depp-orange active:bg-depp-orange active:translate-y-0.5 text-white rounded-xl p-2 w-40 h-auto py-19 text-xs"
            data-testid="registerConfirm"
          >
            グループを設定する
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
